src/strategies/cloudsearch.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+// Load external modules
+const Aws = require('aws-sdk');
+const Twitter = require('twitter-text');
+
+class CloudSearchStrategy {
+  constructor(options) {
+    this._csd = new Aws.CloudSearchDomain(options);
+  }
+
+  uploadDocuments(params) {
+    return new Promise((resolve, reject) => {
+      this._csd.uploadDocuments(params, (err, res) => {
+        if (err) {
+          return reject(err);
+        }
+        resolve(res);
+      });
+    });
+  }
+
+  search(params) {
+    return new Promise((resolve, reject) => {
+      this._csd.search(params, (err, res) => {
+        if (err) {
+          return reject(err);
+        }
+        return resolve(res);
+      });
+    });
+  }
+
+  addDocument(idea) {
+    // Remove certain UTF8 characters from the text
+    const utf8CharsRegex = /[^\u0009\u000a\u000d\u0020-\uD7FF\uE000-\uFFFD]/g;
+    const title = idea.title.replace(utf8CharsRegex, '');
```

Comment:
`title` and `body` are application specific fields, they shouldn't be specified in this module. Same with `idea`.